pkg/api/service/testing/make.go

```diff
@@ -113,6 +114,26 @@ func MakeServicePort(name string, port int, tgtPort intstr.IntOrString, proto ap
 	}
 }
 
+// SetHeadless sets the service as headless and clears other fields.
+func SetHeadless(svc *api.Service) {
+	SetTypeClusterIP(svc)
+	svc.Spec.ClusterIP = api.ClusterIPNone
+}
+
+// SetSelector sets the service selector.
+func SetSelector(sel map[string]string) Tweak {
+	return func(svc *api.Service) {
+		svc.Spec.Selector = sel
```

Comment:
are we fine having selector as a reference, or do we want to protect from undesired mutations like with the svc.Deepcopy()

Reply:
it is fine, all the tests are using it as
```
pkg/registry/core/service/storage/storage_test.go:                              svctest.SetSelector(map[string]string{"k2": "v2"}),
pkg/registry/core/service/storage/storage_test.go:                              svctest.SetSelector(map[string]string{"k2": "v2"}),
```

Reply:
huh.  Good question.  I expect it to mostly be used as `SetSelector(map[string]string{"foo": "bar")), so copying is not important but I could also see wasting time on this in the future.  Will fix in a FIXUP commit.